mmv1/products/dlp/JobTrigger.yaml

```diff
@@ -158,6 +175,413 @@ properties:
+                            required: true
+                            description: Regular expression pattern defining what qualifies
+                              as a hotword.
+                            properties:
+                              - !ruby/object:Api::Type::String
+                                name: 'pattern'
+                                required: true
+                                description: |
+                                  Pattern defining the regular expression. Its syntax
+                                  (https://github.com/google/re2/wiki/Syntax) can be found under the google/re2 repository on GitHub.
+                              - !ruby/object:Api::Type::Array
+                                name: 'groupIndexes'
+                                description: |
+                                  The index of the submatch to extract as findings. When not specified,
+                                  the entire match is returned. No more than 3 may be included.
+                                item_type: Api::Type::Integer
+                          - !ruby/object:Api::Type::NestedObject
+                            name: 'proximity'
+                            required: true
+                            description: |
+                              Proximity of the finding within which the entire hotword must reside. The total length of the window cannot
+                              exceed 1000 characters. Note that the finding itself will be included in the window, so that hotwords may be
+                              used to match substrings of the finding itself. For example, the certainty of a phone number regex
+                              `(\d{3}) \d{3}-\d{4}` could be adjusted upwards if the area code is known to be the local area code of a company
+                              office using the hotword regex `(xxx)`, where `xxx` is the area code in question.
+                            properties:
+                              - !ruby/object:Api::Type::Integer
+                                name: 'windowBefore'
+                                description: |
+                                  Number of characters before the finding to consider. Either this or window_after must be specified
+                              - !ruby/object:Api::Type::Integer
+                                name: 'windowAfter'
+                                description: |
+                                  Number of characters after the finding to consider. Either this or window_before must be specified
+                          - !ruby/object:Api::Type::NestedObject
+                            name: 'likelihoodAdjustment'
+                            required: true
+                            description: |
+                              Likelihood adjustment to apply to all matching findings.
+                            properties:
+                              - !ruby/object:Api::Type::Enum
```

Comment:
exactly_one_of is needed to added to those two fields, fixedLikelihood and relativeLikelihood

Reply:
I found an example for the nested fields with `exactly_one_of` https://github.com/GoogleCloudPlatform/magic-modules/blob/8e1e7c1bb5b69508954413f0c0f193da2527903f/mmv1/products/artifactregistry/Repository.yaml#L238-L239

How did you use exactly_one_of for the sub attributes?


If the API has an validation, I am good to leave them as it is.

Reply:
It works for NestedObject but not in case of NestedArray. Still, I will check if there are any instances where it works for Array, if so will do the same here otherwise we'll rely on the API Error.

Reply:
Currently, `ConflictsWith`, `ExactlyOneOf` or `AtLeastOneOf` are set at the attribute level, which means we can't set it on a list if we have a list of objects, and each object has a list of attributes that's `AtLeastOneOf`. We don't know the length of the list and there is no way to differentiate object[0] from object[1] with what is conflicting. Hence, the use of the above mentioned functions when the attributes are part of array won't work.